app/models/course/reference_time.rb

```diff
@@ -12,6 +12,8 @@ class Course::ReferenceTime < ApplicationRecord
 
   before_destroy :prevent_destroy_if_in_default_timeline, prepend: true
 
+  after_save :reset_closing_reminders, if: :saved_change_to_end_at?
```

Comment:
Wont the reset_closing_reminders be run twice (one from `before_save :reset_closing_reminders, if: :end_at_changed?` in the concern and this one as well?

Reply:
It won't. If it did, then `spec/jobs/course/assessment/closing_reminder_job_spec.rb:10` will fail because it will have enqueued 2 jobs instead of exactly 1.

The double-enqueueing behaviour is prevented by the `!actable_end_at_already_updated` check in `reset_closing_reminders`.
- If the `end_at` change came from the lesson plan item's actable, then `actable_end_at_already_updated` will return `true`, and we don't call `create_closing_reminders_at` from reference time.
- If the `end_at` change came from the reference time (without passing through the lesson plan item's actable), then `actable_end_at_already_updated` will return `false` (since the actable will still have the old reference time's `end_at`), and we call `create_closing_reminders_at` from reference time.

I spotted this behaviour, and it's just nice that it is handled by invariants and tests.

Reply:
Great thanks!

Reply:
Could we add `.exactly(:once)` for the `have_enqueued_job` test cases